tests/integration_test.rs

```diff
@@ -57,3 +57,20 @@ fn struct_struct_association() {
         Err(err) => println!("failure: {:?}", err),
     }
 }
+
+/// Test find_defined_struct() - finds the first populated CPU socket
+#[test]
+fn find_first_cpu() {
+    match table_load_from_device() {
```

Comment:
@jczuluag This shows find_defined_struct() being used and is also a very good case study.

SMBiosProcessorInformation is likely to be difficult for people to use if they have not studied it. The structure doesn't really represent a CPU, it represents a socket for a processor and that processor is not necessarily a CPU, it can be another type of processor.  The most common use for this structure would be to retrieve data about a CPU that is installed on the system (or all CPU's installed on a system).  Two fields need to be checked to ensure you are dealing with a structure representing a socket containing a CPU.

I'm thinking that this knowledge definitely needs to be documented in examples with the structure.  Further, maybe this is the appropriate place to introduce a macro to simplify people's lives by encapsulating the entire logic shown here as if it were a function.